src/test/java/seedu/address/logic/commands/ShowFriendCommandTest.java

```diff
@@ -0,0 +1,87 @@
+package seedu.address.logic.commands;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandFailure;
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandSuccess;
+import static seedu.address.testutil.TypicalIndexes.INDEX_FIRST_PERSON;
+import static seedu.address.testutil.TypicalIndexes.INDEX_SECOND_PERSON;
+import static seedu.address.testutil.TypicalPersons.getTypicalAddressBook;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.person.Name;
+import seedu.address.model.person.Person;
+
+/**
+ * Contains integration tests (interaction with the Model) and unit tests for
+ * {@code ShowFriendCommand}.
+ */
+public class ShowFriendCommandTest {
+
+    private Model model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+
+    //ensures that GUI will only display single person if valid name is given
+    @Test
+    public void execute_showValidFriendName_success() {
+        Person personToShow = model.getFilteredPersonList().get(INDEX_FIRST_PERSON.getZeroBased());
+        ShowFriendCommand showFriendCommand = new ShowFriendCommand(personToShow);
+
+        String expectedMessage = String.format(ShowFriendCommand.MESSAGE_SUCCESS, personToShow.getName());
+
+        ModelManager expectedModel = new ModelManager(model.getAddressBook(), new UserPrefs());
+        expectedModel.updateFilteredPersonList(x -> x.isSamePerson(personToShow));
+
+        assert(expectedModel.getFilteredPersonList().size() == 1);
+
+        assertCommandSuccess(showFriendCommand, model, expectedMessage, expectedModel);
+    }
+
+    @Test
+    public void execute_showNameDoesNotExist_throwsCommandException() {
```

Comment:
Might be worth adding a test case for a filtered list.

Also adding test cases to cover consecutive switching between types. i.e. The `model` should switch from a normal list to an expanded list, and back, if say you do something like `showfriend` then `showfriends` then `showfriend` again, or some sequence like that!